drivers/hmis/app/models/hmis/scan_card_code.rb

```diff
@@ -0,0 +1,25 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::ScanCardCode < Hmis::HmisBase
+  self.table_name = 'hmis_scan_card_codes'
+  acts_as_paranoid
+  has_paper_trail
+
+  belongs_to :client, optional: false, class_name: 'Hmis::Hud::Client'
+  belongs_to :created_by, class_name: 'Hmis::User', optional: true
+  belongs_to :deleted_by, class_name: 'Hmis::User', optional: true
+
+  def self.generate_code
+    'S' + SecureRandom.hex(5).upcase
+  end
+
+  def assign_code
+    return if code.present?
+
+    assign_attributes(code: self.class.generate_code)
+  end
```

Comment:
If I add a uniqueness constraint like @ttoomey suggested, an exception will be thrown and the use wil just have to try again creating one. does that seem ok?

Reply:
Hmm, I guess it's ok, but we don't need to fail, we can just try again until we get one that isn't in use.  Seems odd that you'd get an error on the front end for something you don't control.  If you picked the code manually, I'd feel fine with that, but since we're generating them, we should probably do our best to avoid duplication.

Reply:
https://github.com/greenriver/hmis-warehouse/pull/3923/commits/a4a84b878c62963523b7ad570ec5f345602e78fc

Reply:
Thank you, I think that looks good.